proxy/api/client.go

```diff
@@ -64,6 +65,37 @@ func (client *Client) sendPayload(id string, payload interface{}) (*Response, er
 	return &resp, nil
 }
 
+func (client *Client) sendPayloadGetFd(id string, payload interface{}) (*Response, *os.File, error) {
```

Comment:
I think it would be worth a comment on this function since although the name is fairly descriptive, there are 3 operations going on here (send payload, read fd, read message), but the name suggests 2.